entity-framework/efcore-and-ef6/porting/port-detailed-cases.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Port from EF6 to EF Core - Detailed Cases
+description: Guide to solutions, fixes, and workarounds for common issues that come up during the port from EF 6 to EF Core.
+author: jeremylikness
+ms.alias: jeliknes
+ms.date: 10/25/2021
+uid: efcore-and-ef6/porting/port-detailed-cases
+---
+# Detailed Cases for Porting from EF6 to EF Core
+
+This document details some specific differences between EF6 and EF Core. Consult this guide when porting your code.
+
+## Configuring the database connection
+
+There are several differences between how EF6 connects to various data sources compared to EF Core. They are important to understand when you port your code.
+
+- **Connection strings**: EF Core does not directly support multiple constructor overloads for different connection strings as EF6 does. Instead, it relies on [DbContextOptions](/ef/core/dbcontext-configuration/#dbcontextoptions). You can still provide multiple constructor overloads in derived types, but will need to map connections through the options.
+- **Configuration and cache**: EF Core supports a more robust and flexible implementation of dependency injection with an internal infrastructure that can connect to external service providers. This can be managed by the application to handle situations when the caches must be flushed. The EF6 version was limited and could not be flushed.
+- **Configuration files**: EF6 supports configuration via config files that can include the provider. EF Core requires a direct reference to the provider assembly and explicit provider registration (i.e. `UseSqlServer`).
+- **Connection factories**: EF6 supported connection factories. EF Core does not support connection factories and always requires a connection string.
+- **Logging**: in general, [logging in EF Core](/ef/core/logging-events-diagnostics/) is far more robust and has multiple options for fine-tuned configuration.
+
+## Conventions
+
+EF6 supported custom ("lightweight") conventions and model conventions. The lightweight conventions are similar to EF Core's [pre-convention model configuration](/ef/core/what-is-new/ef-core-6.0/whatsnew#pre-convention-model-configuration). Other conventions are supported as part of model building.
+
+EF6 runs conventions after the model is built. EF Core applies them as the model is being built. In EF Core, you can decouple model building from active sessions with a DbContext, but unlike EF6 you must pass your conventions to the model builder.
```

Comment:
Consider rephrasing the last sentence, also it's possible to get a model builder already initialized with the conventions